src/main/scala/com/fulcrumgenomics/umi/ConsensusCallingIterator.scala

```diff
@@ -50,32 +51,38 @@ class ConsensusCallingIterator[C <: SimpleRead](sourceIterator: Iterator[SamReco
                                threads: Int = 1,
                                maxRecordsInRam: Int = 128000)
   extends Iterator[SamRecord] with LazyLogging {
-  protected val groupingIterator: Iterator[Seq[SamRecord]] = {
-    new SamRecordGroupedIterator(sourceIterator, caller.sourceMoleculeId)
-      .map { records => for (p <- progress; r <- records) p.record(r); records }
+
+
+  private val progressIterator = progress match {
+    case Some(p) => sourceIterator.map { r => p.record(r); r }
+    case None    => sourceIterator
   }
+
+
   protected val iterator: Iterator[SamRecord] = {
     if (threads <= 1) {
+      val groupingIterator = new SamRecordGroupedIterator(progressIterator, caller.sourceMoleculeId)
       groupingIterator.flatMap(caller.consensusReadsFromSamRecords)
     }
     else {
-      val pool          = new ForkJoinPool(threads, ForkJoinPool.defaultForkJoinWorkerThreadFactory, null, true)
-      val bufferedIter  = groupingIterator.bufferBetter
-      val callersFactor = 1000 // the number of callers per thread to create. This to reduce chance of picking the same caller
-      val callers       = Array.range(start=0, threads*callersFactor).map { _ => caller.emptyClone() }
+      val halfMaxRecords   = maxRecordsInRam / 2
+      val groupingIterator = new SamRecordGroupedIterator(new AsyncIterator(progressIterator, Some(halfMaxRecords)).start(), caller.sourceMoleculeId)
```

Comment:
Putting the async iterator here sped up one test case from 5.04 minutes to 4.45 minutes.  Not that it is also not increasing the total number of threads or records in RAM, but "borrowing" from the specified totals.